crates/sui-node/src/lib.rs

```diff
@@ -174,6 +177,16 @@ impl SuiNode {
             .await,
         );
 
+        let checkpoint_executor_handle = {
+            let executor = CheckpointExecutor::new(
+                state_sync_handle.subscribe_to_synced_checkpoints(),
+                checkpoint_store.clone(),
+                state.clone(),
+                CheckpointMetrics::new(&prometheus_registry),
+            )?;
+            spawn_monitored_task!(async move { executor.run().await })
```

Comment:
you probably don't need to use this macro as this will only be called once on node startup, tokio::spawn should be sufficient.

Reply:
Also, since `CheckpointExecutor::run` tasks self by value via a move, you should be able to omit the async block:
```suggestion
            tokio::spawn(executor.run())
```